FluentTc/Locators/GitVCSRootBuilder.cs

```diff
@@ -43,6 +44,7 @@ public interface IGitVCSRootBuilder
         IGitVCSRootBuilder AgentCleanPolicy(AgentCleanPolicy value);
 
         IGitVCSRootBuilder AuthMethod(AuthMethod value);
+        IGitVCSRootBuilder UploadedKey(string value);
```

Comment:
I am not sure I am happy with the method name ```UploadedKey()```. As a user of FluentTc that is familiar with TeamCity SSH Key I would recognize the method purpose if it would be named ```SshKey()```. What do you think?

Reply:
The TeamCity web UI shows "Uploaded Key" as the option available to choose this for auth.  My expectation is that what is on the UI would be most familar to most TeamCity users and that was the reason for the choice.  I'm happy with it either way and will change if you feel strongly about it, though, and don't have a strong preference.

Reply:
I agree with you that it's better to stay closer to the TeamCity UI terminology. Following the same logic, I would expect the AuthMethod also be UploadedKey instead of TeamCitySshKey. What do you think?

Reply:
It's a good question and I'm not sure the best answer.  I didn't really notice it at the time, but the inconsistency is with JetBrains.  The property sent to the API with the name of the key is teamcitySshKey and the AuthMethod property value is TEAMCITY_SSH_KEY.  The way I wrote it, closer to the API and specifying what goes to the API more closely matches the API and closer to a user matches the user interface.  I'd argue that's probably as close to ideal as is expectable, but happy to change it if you think it's better.

Reply:
It's a real dilemma whether the SDK (which FluentTc is) should be closer to user interface (UI) or to programmatic interface (API).  FluentTc was created with a simple vision in mind: it should be easy to discover its API and it should be readable. To fulfill these goals we have to be consistent in naming. So it should be either:

```
var vcsRoot = connectedTc.CreateVcsRoot(__ => __
                .Name("VcsRootName")
                .AuthMethod(AuthMethod.SshKey)
                .SshKey("keyName"); 
```
OR
```
var vcsRoot = connectedTc.CreateVcsRoot(__ => __
                .Name("VcsRootName")
                .AuthMethod(AuthMethod.UploadedKey)
                .UploadedKey("keyName"); 
```
It looks clear that the key refers to AuthMethod.

Looking at both examples above, I recognized that the AuthMethod() is redundant. It cannot specify any other authentication method when UploadedKey was provided. 

So I suggest making this call optional. If user does not call to AuthMethod, it should automatically be set to SshKey/UploadedKey when UploadedKey is provided.

If user calls to AuthMethod() with Anonymous or Password and then provides UploadedKey/SshKey it should throw an exception from the FluentTc and not issue an HTTP request.

Regarding the names I will accept either in sake of consistency.

Reply:
You have convinced me that consistency is of the utmost importance.  My preference would, therefore, to be to use the latter (AuthMethod.UploadedKey and .UploadedKey("keyName")).

However, the translation of the Enum value to the string to send to the remote TeamCity server is done via value.ToString().FromPascalToCapitalizedCase().  For this to work, it needs to be AuthMethod.TeamcitySshKey.

Reply:
Also, you are astute in observing that the choice of AuthMethod can be made implicit.  I'll go ahead and add that.  It will require writing some more tests and making sure I understand intent, so thought I'd put this in first.  Probably worth a separate pull request as it is another enhancement.